rpcd/playbooks/library/git_repo_info

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+# Copyright 2015, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+DOCUMENTATION = """
+---
+module: git_repo_info
+short_description:
+    - A module for gathering git repo facts.
+description:
+    - A module for gathering git repo facts.
+author: Rcbops
+"""
+
+EXAMPLES = """
+- name: Discover current rpc-openstack release
+  git_repo_info:
+    path: "/opt/rpc-openstack"
+  delegate_to: localhost
+  run_once: true
+"""
+
+import os.path
+import subprocess
+
+
+class GitRepoFacts(object):
+    def __init__(self, module):
+        self.state_change = False
+        self.module = module
+        self.params = self.module.params
+
+    def gather_facts(self):
+        """Get information about RPC release."""
+        repo = {}
+        repo['version'], repo['sha'] = self.get_repo_version()
+        repo['name'] = (self.params['name'] or
+                        os.path.split(self.params['path'])[-1])
+        name = '%s_repo' % (repo['name'].replace('-', '_'))
```

Comment:
Could we explicitly set a variable value through Ansible's `register`? This seems a little bit like black magic.